storage/replica_command.go

```diff
@@ -743,6 +743,19 @@ func (r *Replica) PushTxn(batch engine.Engine, ms *engine.MVCCStats, args proto.
 	if err != nil {
 		return reply, err
 	}
+	// There are three cases in which there is no transaction entry, in decreasing likelihood
+	// * a transaction is still active; it just hasn't committed/restarted/aborted/heartbeat yet
+	// * the transaction resolved its intents synchronously on successful commit; in this case,
+	// the transaction record is also removed. Note that in this case, the intent which prompted
+	// this PushTxn doesn't exist any more.
+	// * the transaction timed out/was aborted and the intent not cleaned up, but the transaction
+	// record recycled.
+	// We currently make no attempt at guessing which one it is, though we could (see #1939).
+	// Instead, a new entry is always written, and the Push assumes in all cases that the transaction is still running.
+	// TODO(tschottdorf): we should actually improve this when we garbage-collect aborted transactions,
+	// or we run the risk of a push recreating a GC'ed transaction as PENDING, which is an error if it has open
+	// intents (which is likely if someone pushes it). Likely a non-issue since the timestamp will indicate
+	// the transaction as aborted, yet should be thought out carefully.
```

Comment:
replace
> Likely a non-issue since the timestamp will indicate the transaction as aborted, yet should be thought out carefully.

with

>
This is not a correctness issue since nobody will ever commit this transaction, but
it also updates the intent, leading to more pointless and, potentially cyclic, conflict
resolution. Simply reporting success to the caller and not writing a new entry will
be preferable. See (#insertissuehere).